nextcord/application_command.py

```diff
@@ -3276,3 +3285,179 @@ def unpack_annotation(
+                return value
+
+            def modify(self, option: SlashCommandOption):
+                if self.min and option.min_value is None:
+                    option.min_value = self.min
+                if self.max and option.max_value is None:
+                    option.max_value = self.max
+
+        if isinstance(value, tuple):
+            if len(value) != 2:
+                raise ValueError("Range can only take 1-2 arguments")
+
+            min_value = value[0]
+            max_value = value[1]
+        else:
+            min_value = None
+            max_value = value
+
+        if min_value is None or isinstance(min_value, EllipsisType):
+            Inner.min = None
+        elif isinstance(min_value, (int, float)):
+            Inner.min = min_value
+        else:
+            raise TypeError("Range min must be int or float.")
+
+        if isinstance(max_value, EllipsisType):
+            Inner.max = None
+        elif isinstance(max_value, (int, float)):
+            Inner.max = max_value
+        else:
+            raise TypeError("Range max must be int or float.")
+
+        if Inner.min is None and Inner.max is None:
+            raise TypeError("At least one of min or max must be set.")
+
+        return Inner
+
+
+class String:
+    """An annotation helper for defining slash command `min_length` and `max_length` parameters.
+
+    .. versionadded:: 2.2
+
+    .. container:: operations
+
+        .. describe:: String[x, y]
+
+            Creates a range of string length from ``x`` to ``y``.
+
+        .. describe:: String[x] | String[..., x]
+
+            Create a range of string length up to ``x``.
+
+        .. describe:: String[x, ...]
+
+            Create a range of string length from ``x``.
+    """
+
+    min: ClassVar[Optional[int]]
+    max: ClassVar[Optional[int]]
+
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            Tuple[int, int],
+            Tuple[int, EllipsisType],
+            Tuple[EllipsisType, int],
+        ],
+    ) -> Type[int]:
+        class Inner(String, OptionConverter):
+            def __init__(self):
+                super().__init__(option_type=str)
+
+            async def convert(self, interaction: Interaction, value: Any) -> Any:
+                return value
+
+            def modify(self, option: SlashCommandOption):
+                if self.min and option.min_length is None:
+                    option.min_length = self.min
+                if self.max and option.max_length is None:
+                    option.max_length = self.max
+
+        if isinstance(value, tuple):
+            if len(value) != 2:
+                raise ValueError("String can only take 1-2 arguments")
+
+            min = value[0]
+            max = value[1]
+        else:
+            min = None
+            max = value
+
+        if isinstance(min, EllipsisType):
+            Inner.min = None
```

Comment:
Does this support min being None (if min is left out?) Seems different from Range above.